### PR TITLE
Bulk actions for data node

### LINKED ISSUE #17732: Bulk actions for Data Node restart and removal
Create bulk actions for sequential data node restart and removal

### PR DESCRIPTION
**Pull request by moesterheld**
## Description
Provides functionality for bulk removal/start/stop of data nodes.

Screenshot 2024-02-02 at 15 41 06

## Motivation and Context
resolves #17732 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Review by todvora — APPROVED** [review]
LGTM!

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:35]:
```suggestion
      UserNotification.error(`Removing Data Node failed with status: ${JSON.stringify(failures)}`, 'Could not remove Data Nodes.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:39]:
```suggestion
      UserNotification.success(`${successfully_performed} Data Node${successfully_performed > 1 ? 's' : ''} removed successfully.`);
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:42]:
```suggestion
    UserNotification.error(`Removing Data Node failed with status: ${errorThrown}`, 'Could not remove some Data Nodes.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:55]:
```suggestion
      UserNotification.error(`Starting Data Node failed with status: ${JSON.stringify(failures)}`, 'Could not start Data Nodes.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:59]:
```suggestion
      UserNotification.success(`${successfully_performed} Data Node${successfully_performed > 1 ? 's' : ''} started successfully.`);
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:62]:
```suggestion
    UserNotification.error(`Starting Data Node failed with status: ${errorThrown}`, 'Could not start Data Nodes.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:75]:
```suggestion
      UserNotification.error(`Stopping Data Node failed with status: ${JSON.stringify(failures)}`, 'Could not stop Data Nodes.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:79]:
```suggestion
      UserNotification.success(`${successfully_performed} Data Node${successfully_performed > 1 ? 's' : ''} stopped successfully.`);
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:82]:
```suggestion
    UserNotification.error(`Stopping Data Node failed with status: ${errorThrown}`, 'Could not stop Data Nodes.');
```

**Review by grotlue — APPROVED** [review]
Looks good, just some small typo fixes. Please change before merge. I will approve already as I don't want to block this PR because of the typos.

**Pull request merged**
